src/controllers/ride/searchRides.ts

```diff
@@ -6,18 +6,30 @@ import { z } from "zod";
 import { validate } from "../../helpers/zodValidateRequest";
 
 const dataSchema = z.object({
-  body: z.object({
+  query: z.object({
     fromPlace: z
-      .nativeEnum(Place, {
+    .preprocess(
+      (input) => {
+        const processed = z.string().regex(/^\d+$/).transform(Number).safeParse(input);
```

Comment:
Explain the need for this regex

Reply:
Query params are always recieved as strings. Here fromPlace and toPlace are being verified as nativeEnums of Place, and have to be preprocessed from string to number before validation. And before directly transforming to Number, we need to check if the string itself is of that form. That's what the regex is doing.

Reply:
Is it not possible with zod enums?

Reply:
We're using zod nativeEnum (and not zod enum) everywhere cuz that way the frontend needs to only send a number rather than the string of the whole name, just in case the names of places get too lengthy or complicated in future.